src/hotspot/cpu/x86/c2_MacroAssembler_x86.cpp

```diff
@@ -6234,6 +6234,13 @@ void C2_MacroAssembler::vector_count_leading_zeros_int_avx(XMMRegister dst, XMMR
   vpcmpeqd(xtmp1, xtmp1, xtmp1, vec_enc);
   vpsrld(xtmp1, xtmp1, 24, vec_enc);
 
+  // As 2^24 is the largest possible integer that can be exactly represented by a float value, special handling has to be
```

Comment:
More exactly, +/- 2^24 is the region were all contiguous integers can be represented as floats. All sufficiently large finite floating-point numbers are integers, but the distance between adjacent floating-point numbers is more than 1.

Reply:
Thanks for the clarification! I've adjusted the wording to be more clear.